--- libvirt-cim-0.6.1/Makefile.am.orig	2012-02-08 14:23:43.000000000 +0100
+++ libvirt-cim-0.6.1/Makefile.am	2012-07-12 19:14:39.888957040 +0200
@@ -189,7 +189,7 @@
 	$(install_sh_DATA) -t "$(DESTDIR)$(pkgdatadir)" $(INTEROP_MOFS)
 	$(install_sh_DATA) -t "$(DESTDIR)$(pkgdatadir)" $(INTEROP_REGS)
 	if [[ @CIMSERVER@ != pegasus ]]; then \
-	sed -i '/^# --/,/^# --!/d' $(subst schema,$(DESTDIR)$(pkgdatadir), $(PGINTEROP_REGS)); \
+	sed -i '/^# --/,/^# --!/d' $(subst $(top_srcdir)/schema,$(DESTDIR)$(pkgdatadir), $(PGINTEROP_REGS)); \
 	fi
 
 uninstall-local:
